includes/tracks/events/class-wc-products-tracking.php

```diff
@@ -35,4 +36,37 @@ public static function track_product_updated( $product_id, $post ) {
 
 		WC_Tracks::record_event( 'update_product', $properties );
 	}
+
+	/**
+	 * Send a Tracks event when a product category is created.
+	 *
+	 * @param int $category_id Category ID.
+	 */
+	public static function track_product_category_created( $category_id ) {
+		// phpcs:disable WordPress.Security.NonceVerification.NoNonceVerification
+		// Only track category creation from the edit product screen or the
+		// category management screen (which both occur via AJAX).
+		if (
+			! defined( 'DOING_AJAX' ) ||
+			empty( $_POST['action'] ) ||
+			(
+				// Product Categories screen.
+				'add-tag' !== $_POST['action'] &&
+				// Edit Product screen.
+				'add-product_cat' !== $_POST['action']
+			)
+		) {
```

Comment:
You probably just need to ignore that PHPCS rule here:

```php
) { phpcs:ignore WordPress.Security.NonceVerification.NoNonceVerification
```

Reply:
Or even create a check for `$_POST['action']`, after pass to a variable, will be easier.

Reply:
The way I've currently got it is the only way I've seen it remove the warnings. Here's the output with the suggested ignore comment:

```
FILE: /woocommerce/includes/tracks/events/class-wc-products-tracking.php
----------------------------------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------------------------------
 50 | ERROR | Processing form data without nonce verification.
 53 | ERROR | Processing form data without nonce verification.
 55 | ERROR | Processing form data without nonce verification.
 65 | ERROR | Processing form data without nonce verification.
----------------------------------------------------------------------------------------------
```